src/array/cuda/spmm.cu

```diff
@@ -282,6 +282,107 @@ void CusparseCsrmm2(
   if (valptr)
     device->FreeWorkspace(ctx, valptr);
 }
+
+/*! Cusparse implementation of SpMM on Csr format. */
+template <typename DType, typename IdType>
+void CusparseCsrmm2Hetero(
+    const DLContext& ctx,
+    const CSRMatrix& csr,
+    const DType* B_data, const DType* A_data,
+    DType* C_data,
+    int x_length,
```

Comment:
This should probably be of type `int64_t` so we can check its size internally.

Reply:
Line 293?